### PR TITLE
DEV-53: Create Side Widget

### PR DESCRIPTION
**Pull request by AntonSydorovych**
## Ticket
Link to ticket: https://bsalo.atlassian.net/browse/DEV-53

## Description
Added a Side Widget component

- Side Widget component was copied from Search component and changed;
- Change 2

image

(prefer animated gif)

## Checklist

- [x] Did you test it?
- [x] Did you add tests?
- [x] Did you add to the storybook?

### CODE REVIEW

File: src/components/Side/Side.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.side-widget-card {
+    margin: 20px auto;
+}
+
+.ant-form-inline {
```
Comment (Skitus) [inline src/components/Side/Side.scss:5]: pls don`t use classes of antd , use it with selectors (.side-widget-card .ant-form-inline )

File: src/components/Side/Side.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+const SideWidget = () => {
+  const { handleSubmit } = useForm();
+
+  // eslint-disable-next-line no-console
+  const onSubmit = (data: any) => console.log('Search result', data);
+
+  return (
+    <Card title="Side Widget" className="side-widget-card">
+      <Form onFinish={handleSubmit(onSubmit)} layout="inline">
```
Comment (Skitus) [inline src/components/Side/Side.tsx:14]: Weird. You have to use text inside this card not Form

File: src/components/Side/Side.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import { useForm } from 'react-hook-form';
+
+const SideWidget = () => {
+  const { handleSubmit } = useForm();
+
+  // eslint-disable-next-line no-console
+  const onSubmit = (data: any) => console.log('Search result', data);
+
+  return (
+    <Card title="Side Widget" className="side-widget-card">
```
Comment (Skitus) [inline src/components/Side/Side.tsx:13]: className="..." title="..."

File: src/components/Side/Side.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import './Side.scss';
+import { Card, Form } from 'antd';
+import { useForm } from 'react-hook-form';
+
+const SideWidget = () => {
+  const { handleSubmit } = useForm();
+
+  // eslint-disable-next-line no-console
+  const onSubmit = (data: any) => console.log('Search result', data);
```
Comment (Skitus) [inline src/components/Side/Side.tsx:10]: pls delete this

File: src/stories/Side.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+import SideWidget from '../components/Side/Side';
+
+export default {
+  title: 'Side/SideWidget',
+  component: SideWidget,
+  argTypes: { backgroundColor: { control: 'color' } },
```
Comment (Skitus) [inline src/stories/Side.stories.tsx:8]: delete this pls

File: src/stories/Side.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+} as ComponentMeta<typeof SideWidget>;
+
+const Template: ComponentStory<typeof SideWidget> = () => <SideWidget />;
+
+export const Primary = Template.bind({});
```
Comment (Skitus) [inline src/stories/Side.stories.tsx:13]: delete this

File: src/stories/Side.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+Primary.args = { message: 'message' };
```
Comment (Skitus) [inline src/stories/Side.stories.tsx:14]: same

File: src/components/Side/Side.scss (hunk 1 above)
Comment (Skitus) [reply src/components/Side/Side.scss:5]: add pls color to your header

Comment (Skitus) [reply src/components/Side/Side.scss:5]: Maybe u don`t need this file

File: src/components/Side/Side.scss [hunk 8]
```diff
@@ -0,0 +1,7 @@
+.side-widget-card {
```
Comment (dev1B) [inline src/components/Side/Side.scss:1]: .side-widget достатньо)

File: src/components/Side/Side.scss (hunk 1 above)
Comment (dev1B) [reply src/components/Side/Side.scss:5]: Ізоляція, перепишеш стилі сусіда)

File: src/components/Side/Side.test.tsx [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import SideWidget from './Side';
```
Comment (dev1B) [inline src/components/Side/Side.test.tsx:3]: import SideWidget from './Side';

Side => SideWidget

File: src/components/Side/Side.tsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import './Side.scss';
```
Comment (dev1B) [inline src/components/Side/Side.tsx:2]: src/components/Side/Side.tsx - finename

File: src/components/Side/Side.tsx (hunk 3 above)
Comment (dev1B) [reply src/components/Side/Side.tsx:13]: Краще спершу писати клас а потім все інше, спрощує читання коду)

File: src/pages/Blog/Blog.tsx [hunk 11]
```diff
@@ -5,6 +5,7 @@ import Pagination from '../../components/Pagination';
 import './Blog.scss';
 import PostsCardList from '../../components/BlogPosts/PostCardList';
 import SearchWidget from '../../components/Search/Search';
+import SideWidget from '../../components/Side/Side';
```
Comment (dev1B) [inline src/pages/Blog/Blog.tsx:8]: import SideWidget from '../../components/Side/Side';

`/Side/Side';`

File: src/stories/Side.stories.tsx (hunk 5 above)
Comment (dev1B) [reply src/stories/Side.stories.tsx:8]: @Skitus А ти не видаляєш)))

File: src/stories/Side.stories.tsx (hunk 7 above)
Comment (dev1B) [reply src/stories/Side.stories.tsx:14]: `const SideWidget = () => {`

твій компонент ніц не приймає)

**Review by dev1B — CHANGES_REQUESTED** [review]

File: src/components/SideWidget/SideWidget.tsx [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import './SideWidget.scss';
+import { Card } from 'antd';
+
+const SideWidget = () => {
+  const text = `You can put anything you want inside
```
Comment (dev1B) [inline src/components/SideWidget/SideWidget.tsx:6]: remove `const text`

File: src/stories/SideWidget.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+import SideWidget from '../components/SideWidget/SideWidget';
+
+export default {
+  title: 'Widget/SideWidget',
```
Comment (dev1B) [inline src/stories/SideWidget.stories.tsx:6]: Widget => Widgets ?

**Review by dev1B — CHANGES_REQUESTED** [review]

**Pull request merged**